src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('.employees-table');
+
+  if (!table) {
+    return;
+  }
+
+  const headers = table.querySelectorAll('th');
+  const rows = Array.from(table.querySelectorAll('tbody tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const isNumeric = header.dataset.type === 'number';
+      const sortedRows = rows.sort((rowA, rowB) => {
+        const cellA = rowA.children[index].textContent.trim();
+        const cellB = rowB.children[index].textContent.trim();
+
+        if (isNumeric) {
+          return parseFloat(cellA) - parseFloat(cellB);
+        } else {
+          return cellA.localeCompare(cellB);
+        }
+      });
```

Comment:
Ensure that the `rows` array is redefined or updated each time a header is clicked. Currently, the `rows` array is defined once outside the event listener, which means it won't reflect any changes made to the table after the initial load. Consider moving the definition of `rows` inside the click event listener to ensure it always reflects the current state of the table.